src/Vendr.PaymentProviders.Stripe/StripeCheckoutPaymentProvider.cs

```diff
@@ -0,0 +1,563 @@
+                if (!string.IsNullOrWhiteSpace(stripePaymentIntentId))
+                {
+                    var secretKey = settings.TestMode ? settings.TestSecretKey : settings.LiveSecretKey;
+
+                    ConfigureStripe(secretKey);
+
+                    var paymentIntentService = new PaymentIntentService();
+                    var intent = paymentIntentService.Cancel(stripePaymentIntentId);
+
+                    return new ApiResult()
+                    {
+                        TransactionInfo = new TransactionInfoUpdate()
+                        {
+                            TransactionId = GetTransactionId(intent),
+                            PaymentStatus = GetPaymentStatus(intent)
+                        }
+                    };
+                }
+
+                // If there is a charge, then it's too late to cancel
+                // so we attempt to refund it instead
+                var chargeId = order.Properties["stripeChargeId"];
+                if (chargeId != null)
+                    return RefundPayment(order, settings);
+            }
+            catch (Exception ex)
+            {
+                Vendr.Log.Error<StripeCheckoutOneTimePaymentProvider>(ex, "Stripe - CancelPayment");
+            }
+
+            return ApiResult.Empty;
+        }
+
+        public override bool CanProcessOrder(OrderReadOnly order, StripeCheckoutSettings settings, ref string errorMessage)
+        {
+            long recurringTotalPrice = 0;
+            bool hasRecurringItems = false;
+            long orderTotalPriceWithPaymentMethodFee = AmountToMinorUnits(order.TotalPrice.Value.WithTax - order.PaymentInfo.TotalPrice.Value.WithTax);
+
+            var lineItems = new List<SessionLineItemOptions>();
+
+            foreach (var orderLine in order.OrderLines.Where(IsRecurringOrderLine))
+            {
+                recurringTotalPrice += AmountToMinorUnits(orderLine.TotalPrice.Value.WithTax);
+                hasRecurringItems = true;
+            }
+
+            // If we don't have any recurring items then we can process the order fine
+            if (!hasRecurringItems)
+                return true;
+
+            // If we do have recurring items, make sure the total price of the order
+            // is greater than the value of all recurring order items
+            if (recurringTotalPrice <= orderTotalPriceWithPaymentMethodFee)
+                return true;
+
+            errorMessage = "Cannot process orders where the total value of recurring order items is greater than the order total";
+
+            return false;
+        }
+
+        private bool IsRecurringOrderLine(OrderLineReadOnly orderLine)
+        {
+            return orderLine.Properties.ContainsKey("isRecurring")
+                && (orderLine.Properties["isRecurring"] == "1" 
+                    || orderLine.Properties["isRecurring"] == "true" 
+                    || orderLine.Properties["isRecurring"] == "True");
```

Comment:
@mattbrailsford  maybe it should use this instead?

```
orderLine.Properties["isRecurring"]?.Value?.Equals("True", StringComparison.OrdinalIgnoreCase)
```

or it could use `TryConvertTo<bool>()` from Umbraco core which would return success and `true`  for both `1`, `true` and `True`.

Reply:
I've tried to avoid payment providers having a hard Umbraco dependency so I've gone for something more along the lines of the original suggestion 👍

Reply:
Maybe Vendr should have a constant for the `isRecurring` similar to the ones for customer first name, last name and email? Would `isPaymentRecurring` more meaningful? Not sure if ou might want something else on the order line to track something "is recurring"?

Reply:
I have added it in to core as a constant, but I don't want to push it's use yet until we have some other payment providers that support subscriptions as right now we only know how stripe works so I want to make sure whatever we come up with works for others too.

I think "isRecurring" is fine as it's descriptive of the product. We have "isGiftCard" to state if a product is a gift card, so I think "isRecurring" is descriptive that the product is a recurring product.